utility/doc/PROTOCOL_SESSION.md

```diff
@@ -1,81 +1,54 @@
-3. Get an ordered list of the public keys of servicers who are:
-   - actively staked
-   - staked within geo-zone
-   - staked for relay-chain
-4. Pseudo-insert the session `key` string into the list and find the first actor directly below on the list
-5. Determine a new seedKey with the following formula: ` key = Hash( key + actor1PublicKey )` where `actor1PublicKey` is the key determined in step 4
-6. Repeat steps 4 and 5 until all N servicers are found
-7. Do steps 3 - 6 for Fishermen as well
+_WIP: Run `TODO_SEARCH=utility/session_ make todo*search` to identify all the WIP related to session generation.*
 
-### FAQ
+## Interface
 
-- Q) why do we hash to find a newKey between every actor selection?
-- A) pseudo-random selection only works if each iteration is re-randomized or it would be subject to lexicographical proximity bias attacks
+`Pocket` satisfies the V1 Utility Specification's Session Protocol by implementing the following function from the [Utility Module Interface](../../shared/modules/utility_module.go) and returning a [Session Protobuf](../../shared/core/types/proto/session.proto):
 
-- Q) Why do we not use Golang's `rand.Intn` with the key as a seed for random node selection?
-- A) A proprietary randomization algorithm makes this approach language & library agnostic, so any client simply has to follow the specifications
+```go
+GetSession(appAddr string, sessionHeight int64, relayChain string, geoZone string) (*coreTypes.Session, error)
+```
 
-- Q) what is `WorldState`?
-- A) it represents a queryable view on the internal state of the network at a certain height.
+## Session Creation Flow
 
-- Q) Do Fishermen stake for a specific RelayChain?
-- A) Fishermen are only going to be applicable to Pocket Supported Relay Chains (where the protocol pays out for the relay chain). It is unclear at this time what the limitations and scoping will be for Fishermen RelayChain support.
+The following is a simplified flow of the session creation flow for illustrative purposes only.
 
-- Q) What was the reasoning not to allow a list of geozones?
-- A) Each session is mono-chain and mono-geo. This is fundamental as it would create even more possible combinations of sessions and increase computational complexity during block production and servicing
+See [session.go](../session.go) and [session_test.go](../session_test.go) for the full implementation.
 
-### Session Flow
+1. Create a session object from the seed data
+2. Create a key concatenating and hashing the seed data
+   - `sessionId = Hash(sessionHeight + blockHash + geoZone + relayChain + appPublicKey)`
+3. Get an ordered list of the public keys of servicers and fishermen who are:
+   - actively staked
+   - staked within geo-zone
+   - staked for relay-chain
+4. Use a pseudo-random selection algorithm to retrieve the fishermen and servicers for for the sessionId
 
 ```mermaid
 sequenceDiagram
     autonumber
```

Comment:
I'm noticing that the sequence numbers don't match up anymore. What do you think about dropping them and/or adding a heading to introduce the diagram section and/or distinguish it from the "simplified flow" outlined above?

Reply:
bump